util/src/org/commcare/util/cli/EntityScreen.java

```diff
@@ -113,10 +114,17 @@ public boolean setCurrentScreenToDetail() {
         return true;
     }
 
-    public void setCurrentScreenToDetail(int index) {
+    public void setCurrentScreenToDetail(int index) throws CommCareSessionException {
         EvaluationContext subContext = new EvaluationContext(mSession.getEvaluationContext(), this.mCurrentSelection);
 
-        this.mCurrentScreen = new EntityDetailSubscreen(index, this.mLongDetailList[index], subContext, getDetailListTitles(subContext));
+        TreeReference detailNodeset = this.mLongDetailList[index].getNodeset();
+        if (detailNodeset != null) {
+            TreeReference contextualizedNodeset = detailNodeset.contextualize(this.mCurrentSelection);
+            this.mCurrentScreen = new EntityListSubscreen(this.mLongDetailList[index], subContext.expandReference(contextualizedNodeset), subContext);
```

Comment:
Yeah, wanting this to actually work in the CLI is reasonable. Don't want it to block this PR, but happy to make myself a ticket and do it once I'm through my near-term TDH tasks.